query/src/storages/fuse/io/block_reader.rs

```diff
@@ -88,67 +83,50 @@ impl BlockReader {
             &metadata.row_groups[0]
         };
 
-        let col_num = self.projection.len();
-        let cols = self
+        let arrow_fields = &self.arrow_table_schema.fields;
+        let stream_len = self.file_len;
+        let read_buffer_size = self.read_buffer_size;
+        let parquet_fields = metadata.schema().fields();
+
+        // read_columns_many_async use field name to filter columns
+        let fields_to_read = self
             .projection
             .clone()
             .into_iter()
-            .map(|idx| (row_group.column(idx).clone(), idx));
-
-        let fields = self.table_schema.fields();
-        let arrow_fields = self.arrow_table_schema.fields();
-        let stream_len = self.file_len;
-        let read_buffer_size = self.read_buffer_size;
+            .map(|idx| {
+                let origin = arrow_fields[idx].clone();
+                Field {
+                    name: parquet_fields[idx].name().to_string(),
+                    ..origin
+                }
+            })
+            .collect();
 
-        let stream = futures::stream::iter(cols).map(|(col_meta, idx)| {
+        let factory = || {
```

Comment:
we use the arrow2 parallel read_columns_many_async here

Reply:
@Xuanwo  (data_accessor, path) ?

Reply:
some thoughts here:

1. shall we control the number of futures that simultaneously read the data of columns?
2. about the `try_join_all` used in `read_columns_may_async`
    at least, "join_all will switch to the more powerful [FuturesOrdered](https://docs.rs/futures/0.3.21/futures/stream/struct.FuturesOrdered.html) for performance reasons" according to 
     https://docs.rs/futures/0.3.21/futures/future/fn.join_all.html#see-also
     not sure if it's worth bothering to replace it.
     
 3. read column chunk in one go might be a huge benefit (as @youngsofun already mentioned)

   now the `read_page_header` scattered read bytes in an in-memory `Cursor`
 https://github.com/jorgecarleitao/parquet2/blob/85d1f01597907c0cc30a234a6d6209c3d7ef17cf/src/read/page_iterator.rs#L65-L68
 
  maybe we can eliminate the `BufReader` used in the reading of column
    
    https://github.com/datafuselabs/databend/blob/b41d39b05dcf281db921377d8b027477d333f8b6/query/src/storages/fuse/io/block_reader.rs#L108-L111

Reply:
cursor is only for file metadata https://github.com/jorgecarleitao/parquet2/search?q=cursor

but we can eliminate the BufReader because "we execute exactly 1 seek and 1 read on them."  https://github.com/jorgecarleitao/arrow2/blob/3d528c99589e96f0539de4c07b11843fa22f23ac/examples/parquet_read_async.rs#L31

Reply:
> cursor is only for file metadata

Oh, I mean the `Cursor` used in arrow2 `to_deserializer`

https://github.com/jorgecarleitao/arrow2/blob/3d528c99589e96f0539de4c07b11843fa22f23ac/src/io/parquet/read/row_group.rs#L173-L185

> pr https://github.com/datafuselabs/databend/pull/4230

:+1: